main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -1401,4 +1408,45 @@ object Type {
     case Constant.RecordEmpty => Type.mkRecord(Type.RecordRowEmpty, SourceLocation.Unknown)
   }
 
+  /**
+    * Replaces the given region in the type with the Pure effect.
+    */
+  def purifyRegion(tpe0: Type, sym: Symbol.RegionSym): Type = tpe0 match {
+    case Cst(TypeConstructor.Region(`sym`), _) => Type.Pure
```

Comment:
Please don't do this.

Reply:
![image](https://github.com/user-attachments/assets/cd5f3f0a-9aad-4f61-aabf-c38ed6843bb6)